### PR TITLE
Add missing word to ServiceSpec doc

### PR DESCRIPTION
**Pull request by brpratt**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

The description for `externalName` in [ServiceSpec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#servicespec-v1-core) ends rather abruptly:

> Must be a lowercase RFC-1123 hostname (https://tools.ietf.org/html/rfc1123) and requires Type to be

This PR adds the missing `ExternalName` to the end of this description.

> Must be a lowercase RFC-1123 hostname (https://tools.ietf.org/html/rfc1123) and requires Type to be **ExternalName**

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -4150,7 +4150,7 @@ type ServiceSpec struct {
 	// externalName is the external reference that discovery mechanisms will
 	// return as an alias for this service (e.g. a DNS CNAME record). No
 	// proxying will be involved.  Must be a lowercase RFC-1123 hostname
-	// (https://tools.ietf.org/html/rfc1123) and requires Type to be
+	// (https://tools.ietf.org/html/rfc1123) and requires Type to be ExternalName
```
Comment (tengqm) [inline staging/src/k8s.io/api/core/v1/types.go:4153]:
```suggestion
	// (https://tools.ietf.org/html/rfc1123) and requires Type to be "ExternalName".
```

Comment (brpratt) [reply staging/src/k8s.io/api/core/v1/types.go:4153]: I will make the suggested change and regenerate the other files. 

Is there a convention doc I can reference that explains how to annotate these comments? From the suggestion, it appears as though field names should be surrounded by backticks and field values are surrounded by double quotes. I noticed that this convention isn't followed for many other field descriptions.

For example, should similar changes be made in the comment for the `HealthCheckNodePort` field in ServiceSpec?

Comment (tengqm) [reply staging/src/k8s.io/api/core/v1/types.go]: The rule of thumb is that the comments are used to generate user-facing documents. Users will see field names encoded in the JSON (OpenAPI spec), not the Golang field name. That is the reason we need to change "Type" to `type`. The ExternalName is a string value to be assigned to the `type` field, adding the double quotes is for clarification.

File: staging/src/k8s.io/api/core/v1/types.go [hunk 2]
```diff
@@ -4150,7 +4150,7 @@ type ServiceSpec struct {
 	// externalName is the external reference that discovery mechanisms will
 	// return as an alias for this service (e.g. a DNS CNAME record). No
 	// proxying will be involved.  Must be a lowercase RFC-1123 hostname
-	// (https://tools.ietf.org/html/rfc1123) and requires Type to be
+	// (https://tools.ietf.org/html/rfc1123) and requires `type` to be "ExternalName".
```
Comment (aojea) [inline staging/src/k8s.io/api/core/v1/types.go:4153]: I couldn't´t find any other place that uses backticks for `tyoe`

Comment (brpratt) [reply staging/src/k8s.io/api/core/v1/types.go:4153]: I added backticks per this suggestion:

https://github.com/kubernetes/kubernetes/pull/97896#discussion_r554985889

Indeed, I can't find many usages of backticks in this file, but I didn't know if using backticks to annotate field names was a new convention.

Comment (aojea) [reply staging/src/k8s.io/api/core/v1/types.go:4153]: ups sorry, I missed it, my fault

**Pull request merged**